xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -1835,7 +1836,29 @@ CDemuxStream* CDVDDemuxFFmpeg::AddStream(int streamIdx)
       }
     }
     if (langTag)
-      stream->language = std::string(langTag->value, 3);
+    {
+      std::string langCode{langTag->value, 3};
+      //! @FIXME: Matroska v4 support BCP-47 language code with LanguageIETF element
+      //! that have the priority over the Language element, but this is not currently
+      //! implemented in to ffmpeg library. Since ffmpeg read only the Language element
+      //! all tracks will be identified with same language (of Language element).
+      //! As workaround to allow set the right language code we provide the possibility
+      //! to set the language code in the title field in the following format:
+      //! "(zh-tw)" or "(zh_hans)"
+      //! this allow to kodi to recognize the right language and select the
+      //! right track to be played at playback starts.
+      AVDictionaryEntry* title = av_dict_get(pStream->metadata, "title", NULL, 0);
+      if (title && title->value)
+      {
+        CRegExp regLangCode;
```

Comment:
I'd say this logic here shouldn't be part of the demuxer cop file but rather as some static method in utils (maybe language code expander). The simple fact you include low level stuff like regex is a clear sign of this

Reply:
done